### PR TITLE
Fix bearings calculation when reroute

### PR DESCRIPTION
**Pull request by Lebedinsky**
### Description
Fix bearings calculation when reroute

### Changelog
```
<changelog>Fixed bearings calculation when reroute happened</changelog>
```

### CODE REVIEW

**Comment by Lebedinsky** [comment]
This PR contains some example screen that shouldn't be merged. It was added just for testing purposes. So after review of this PR, we need to remove all example code for this fix.

**Review by Guardiola31337 — COMMENTED** [review]
There are currently a couple of tests (`MapboxRouteOptionsUpdaterTest`) failing https://app.circleci.com/pipelines/github/mapbox/mapbox-navigation-android/10991/workflows/53422369-eb52-462d-87f5-f9187e58a44f/jobs/43455 Can we fix them and add any others needed covering the changes made as part of this PR?

Also, noting that this needs to be cherry-picked into the `release-v1.x` branch - added `needs-backporting` label so we don't forget 😅

**Comment by RingerJK** [comment]
@Lebedinsky could you fix CI failing and add tests for new bearing logic?

File: libnavigation-core/src/main/java/com/mapbox/navigation/core/routeoptions/MapboxRouteOptionsUpdater.kt [hunk 1]
```diff
@@ -146,13 +146,13 @@ class MapboxRouteOptionsUpdater(
             if (legacyBearingList != null) {
                 newList.addAll(
                     legacyBearingList.subList(
-                        legIndex + 1,
+                        coordinates - remainingWaypoints,
```
Comment (kmadsen) [inline libnavigation-core/src/main/java/com/mapbox/navigation/core/routeoptions/MapboxRouteOptionsUpdater.kt:149]: @Lebedinsky can you add/modify a test in `MapboxRouteOptionsUpdaterTest`? it should fail before this change, and pass after this change.

**Comment by korshaknn** [comment]
@Lebedinsky do we need test activities? can they be removed to simplify the review?

**Comment by Lebedinsky** [comment]
> @Lebedinsky do we need test activities? can they be removed to simplify the review?

Yes, agree, this activity was just added to check everything is working and actually they shouldn't be as part of this fix. Thanks @korshaknn

**Review by Guardiola31337 — COMMENTED** [review]
Noting that removing `MapboxVoiceActivity` and `mapbox_access_token.xml` is being handled in https://github.com/mapbox/mapbox-navigation-android/pull/4198

**Comment by RingerJK** [comment]
mostly looks great. Could you also add different `remainingWaypoints` as a parameter of the test(should be injected via constructor like 
```
class BearingOptionsParameterized(
        val routeOptions: RouteOptions,
        val expectedBearings: List<List<Double>?>,
        val remainingWaypoints: Int
    )
```

this param should change
https://github.com/mapbox/mapbox-navigation-android/blob/bab400e46b09f7848090236966f4e524672ee9d7/libnavigation-core/src/test/java/com/mapbox/navigation/core/routeoptions/MapboxRouteOptionsUpdaterTest.kt#L269

**Comment by RingerJK** [comment]
@Lebedinsky thanks! Add please corner cases as well: `remainingWaypoints` is `0`, `4`(the same number as coordinates in test) 🙏

**Comment by Lebedinsky** [comment]
> @Lebedinsky thanks! Add please corner cases as well: `remainingWaypoints` is `0` and`4`(the same number as coordinates in test) 🙏

Case with `remainingWaypoints == 0` covered with test `new_options_invalid_remaining_points`. I don't think we need to cover case with `4`, it is impossible on our side and should be covered on NN side.

**Review by RingerJK — APPROVED** [review]
lgtm 👍  please squash and rebase before merge
